idaes/models/properties/general_helmholtz/helmholtz_functions.py

```diff
@@ -481,6 +634,31 @@ def _sv_str(**kwargs):
     def add_funcs(self, names=None):
         add_helmholtz_external_functions(self.blk, names=names)
 
+    def _state_vars(self, **kwargs):
+        c = self.param.pure_component
+        blk = self.blk
+        if "p" in kwargs and kwargs["p"] is not None:
+            p = kwargs["p"] * self.param.uc["Pa to kPa"]
+            if "h" in kwargs and kwargs["h"] is not None:
+                if self.amount_basis == AmountBasis.MOLE:
+                    h = kwargs["h"] * self.param.uc["J/mol to kJ/kg"]
+                else:
+                    h = kwargs["h"] * self.param.uc["J/kg to kJ/kg"]
+                return StateVars.PH, blk, c, h, p
+            if "s" in kwargs and kwargs["s"] is not None:
+                if self.amount_basis == AmountBasis.MOLE:
+                    s = kwargs["s"] * self.param.uc["J/mol/K to kJ/kg/K"]
+                else:
+                    s = kwargs["s"] * self.param.uc["J/kg/K to kJ/kg/K"]
+                return StateVars.PS, blk, c, s, p
+            if "u" in kwargs and kwargs["u"] is not None:
+                if self.amount_basis == AmountBasis.MOLE:
+                    u = kwargs["u"] * self.param.uc["J/mol to kJ/kg"]
+                else:
+                    u = kwargs["u"] * self.param.uc["J/kg to kJ/kg"]
```

Comment:
All these are indented a level beyond what they should, I think

Reply:
I think it's right they are in the "if p is an args."  I will double check everything in the function again though.

Reply:
So all of those conversions should happen only if pressure is a state variable and also converted?

Reply:
I used to have (s, T), but as it is pressure is always a state variable. Could change in the future if we find a reason.